ignite/pkg/xgenny/runner.go

```diff
@@ -0,0 +1,140 @@
+package xgenny
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/gobuffalo/genny/v2"
+
+	"github.com/ignite/cli/v28/ignite/pkg/placeholder"
+	"github.com/ignite/cli/v28/ignite/pkg/randstr"
+	"github.com/ignite/cli/v28/ignite/pkg/xos"
+)
+
+type Runner struct {
+	*genny.Runner
+	Path     string
+	ctx      context.Context
+	tracer   *placeholder.Tracer
+	TempPath string
+}
+
+// NewRunner is a xgenny Runner with a logger.
+func NewRunner(ctx context.Context, appPath string) *Runner {
+	var (
+		runner  = genny.WetRunner(ctx)
+		tmpPath = filepath.Join(os.TempDir(), randstr.Runes(5))
+	)
+	r := &Runner{
+		ctx:      ctx,
+		Runner:   runner,
+		Path:     appPath,
+		TempPath: tmpPath,
+		tracer:   placeholder.New(),
+	}
+	runner.FileFn = func(f genny.File) (genny.File, error) {
+		return wetFileFn(r, f)
+	}
+	return r
+}
+
+func (r *Runner) Tracer() *placeholder.Tracer {
+	return r.tracer
+}
+
+// ApplyModifications copy all modifications from the temporary folder to the target path.
+func (r *Runner) ApplyModifications() (SourceModification, error) {
+	sm := NewSourceModification()
+	if _, err := os.Stat(r.TempPath); os.IsNotExist(err) {
+		return sm, nil
+	}
+
+	// Create the target path and copy the content from the temporary folder.
+	if err := os.MkdirAll(r.Path, os.ModePerm); err != nil {
+		return sm, nil
+	}
+	err := xos.CopyFolder(r.TempPath, r.Path)
+	if err != nil {
+		return sm, nil
+	}
+
+	// fetch the source modification
+	for _, file := range r.Results().Files {
+		fileName := file.Name()
+		_, err := os.Stat(fileName)
```

Comment:
@Pantani I'm a bit confused with this result file path, shouldn't it be checked for existence after the files from the temporary folder are copied to the destination path?

Reply:
totally right!! thanks for catching this one